crates/rooch-rpc-server/src/server/rooch_server.rs

```diff
@@ -393,24 +395,48 @@ impl RoochAPIServer for RoochServer {
             let start_sub = start
                 .checked_sub(limit_of)
                 .ok_or(jsonrpsee::core::Error::Custom(
-                    "cursor value overflow".to_string(),
+                    "cursor value is overflow".to_string(),
                 ))?;
             let end = if start >= limit_of { start_sub } else { 0 };
+
+            let gaps = end
+                .checked_sub(start)
+                .ok_or(jsonrpsee::core::Error::Custom(
+                    "end value is overflow".to_string(),
+                ))?;
+            if gaps > MAX_RESULT_AMOUNT {
+                return Err(jsonrpsee::core::Error::Custom(
+                    "end value is overflow".to_string(),
+                ));
+            }
```

Comment:
We just need to check the limit argument is not too big.

Reply:
And `10000000` is too big, maybe `10000`?

Reply:
Fixed

Reply:
```        let limit_of = min(
            limit.map(Into::into).unwrap_or(DEFAULT_RESULT_LIMIT_USIZE),
            MAX_RESULT_LIMIT_USIZE,
        );
```
It is enough to add this check to `get_transactions_by_order`, and other overflow judgments about the limit are not needed.
Other methods are the same as above.

Reply:
Use the code of @baichuan3

Reply:
Fixed